stubs/mu-plugin.php

```diff
@@ -0,0 +1,144 @@
+            self::$cache = self::RedisObjectCache;
+
+            $client = class_exists('Redis') ? 'phpredis' : 'predis';
+            self::$client = defined('WP_REDIS_CLIENT') ? str_replace('pecl', 'phpredis', WP_REDIS_CLIENT) : $client;
+        }
+
+        if (method_exists($class, '_connect_redis')) {
+            self::$cache = self::WpRedis;
+            self::$client = 'phpredis';
+        }
+
+        if (property_exists($class, '_object_cache')) {
+            self::$cache = self::LiteSpeedCache;
+            self::$client = 'phpredis';
+        }
+    }
+
+    public static function print(): void
+    {
+        if (! self::$cache) {
+            return;
+        }
+
+        if (! self::shouldPrint()) {
+            return;
+        }
+
+        printf(
+            "\n<!-- plugin=%s client=%s %s -->\n",
+            self::$cache,
+            self::$client,
+            self::getMetrics()
+        );
+    }
+
+    protected static function shouldPrint(): bool
+    {
+        return ! ((defined('\WP_CLI') && constant('\WP_CLI')) ||
+                  (defined('\REST_REQUEST') && constant('\REST_REQUEST')) ||
+                  (defined('\XMLRPC_REQUEST') && constant('\XMLRPC_REQUEST')) ||
+                  (defined('\DOING_AJAX') && constant('\DOING_AJAX')) ||
+                  (defined('\DOING_CRON') && constant('\DOING_CRON')) ||
+                  (defined('\DOING_AUTOSAVE') && constant('\DOING_AUTOSAVE')) ||
+                  (function_exists('wp_is_json_request') && wp_is_json_request()) ||
+                  (function_exists('wp_is_jsonp_request') && wp_is_jsonp_request()));
+    }
+
+    protected static function getMetrics(): string
+    {
+        global $wp_object_cache;
+
+        switch (self::$cache) {
+            case self::ObjectCachePro:
+                return $wp_object_cache->requestMeasurement();
+            case self::RedisObjectCache:
+                $info = $wp_object_cache->info();
+
+                return self::buildMetrics(
+                    $info->hits,
+                    $info->misses,
+                    $info->ratio,
+                    $info->bytes
+                );
+            case self::WpRedis:
+                return self::buildMetrics(
+                    $wp_object_cache->cache_hits,
+                    $wp_object_cache->misses,
+                    99, // TODO: Calculate WP Redis ratio
+                    9999 // TODO: Calculate WP Redis bytes
+                );
+            case self::LiteSpeedCache:
+                return self::buildMetrics(
+                    $wp_object_cache->count_hit,
+                    $wp_object_cache->count_miss,
+                    99, // TODO: Calculate LiteSpeed Cache ratio
+                    9999 // TODO: Calculate LiteSpeed Cache bytes
+                );
+        }
+
+        return '';
+    }
+
+    protected static function buildMetrics($hits, $misses, $ratio, $bytes): string
```

Comment:
We can calculate the bytes using `$this->cache` or whatever using the reflection API and then calling `strlen(serialize($cache))`.